src/index.js

```diff
@@ -1,567 +1,363 @@
- * @return {Promise => ArrayBuffer} Promise that resolves to arrayBuffer
- */
-const hash = async (...args) => {
-  // map args into buffers and concat into one buffer
-  const buffer = Buffer.concat(args.map(a => Buffer.from(a)));
-  // generate a SHA-512 hash from input buffer
-  return crypto.subtle.digest({ name: 'SHA-512' }, buffer);
-};
-
+    .join('')
+}
 
 
 /**
- * Runs argon2wasm to return a seed of desired bytes
- * @param  {Uint8Array, Buffer, string} entropy ticket bytes as string or
- * Uint8Array or Buffer, at least 16 bytes
- * @param  {int} seedSize desired size of the generated seeds in bytes
- * @return {Promise => ArrayBuffer} Promise that resolves to arrayBuffer
+ * Derive a 256-bit key from provided entropy via Argon2.
+ *
+ * @param  {String}  entropy
+ * @return {Promise<Object>} derived key
  */
-const argon2u = (entropy, seedSize) => argon2.hash({
-  pass: entropy, // string or Uint8Array
+const argon2u = entropy => argon2.hash({
+  pass: entropy,
   salt: 'urbitkeygen',
   type: argon2.types.Argon2u,
-  hashLen: seedSize,
-  // distPath: 'node_modules/argon2-wasm/dist',
+  hashLen: 32,
   parallelism: 4,
   mem: 512000,
   time: 1,
-});
-
+})
 
+/**
+ * SHA-256 hash function.
+ *
+ * @param  {Array, ArrayBuffer, Buffer, String} args any number of arguments
+ * @return {Promise<ArrayBuffer>}  the hash, as bytes
+ */
+const sha256 = async (...args) => {
+  const buffer = Buffer.concat(args.map(Buffer.from))
+  return crypto.subtle.digest({ name: 'SHA-256' }, buffer)
+}
 
 /**
- * Derive a new seed from a seed. Uses a config with the following entries:
- * @param  {Buffer}   seed seed to derive from.
- * @param  {string}   type the type of the seed we want to derive:
- * ("transfer", "spawn", "voting", "manage", "network").
- * @param  {object}   revision the revision number of the seed we want to derive.
- * @param  {integer}  ship  optional ship number we want to derive the seed for.
- * @param  {string}   password  optional password to salt the seed with before
- * deriving.
- * @return {Promise => Buffer} a new seed
+ * Derive a child seed from a parent.
+ *
+ * @param  {Array, ArrayBuffer, Buffer, String}  seed a parent seed
+ * @param  {String}  type the type of child seed to derive
+ * @param  {Number}  ship the ship to derive the seed for
+ * @param  {Number}  revision the revision number
+ * @return {Promise<String>} the BIP39 child mnemonic
  */
 const childSeedFromSeed = async config => {
-  const { seed, type, revision, ship, password } = config;
-
-  const salt = isNumber(ship)
-    ? `${type}-${revision}-${ship}`
-    : `${type}-${revision}`;
-
-  const childSeed = await hash(seed, salt, defaultTo(password, ''));
-
-  return childSeed.slice(0, seed.byteLength || seed.length);
-};
+  const { seed, type, ship, revision } = config
+  const salt = lodash.isNull(ship) ? '' : `${ship}`
+  const hash = await sha256(seed, type, salt, `${revision}`)
```

Comment:
This isn't doing dashes in between the salt parameters. It won't distinguish between ship `5` with revision `10`, and ship `51` with revision `0`.

Pretty sure we want to do `type-revision-ship` like we used to, and then use `0` as the defaults for anything not supplied. Seems correct, @msutherl ?

Reply:
Yeah, agree that's better.  But we should specify the salt format concretely in the spec, which at present reads along the lines of what I've done here.  /willfix in the meantime.